src/api/libcellml/component.h

```diff
@@ -74,16 +76,32 @@ class LIBCELLML_EXPORT Component: public ComponentEntity
     void addVariable(const VariablePtr &v);
 
     /**
-     * @brief Remove the variable with the given name from this component.
+     * @brief Remove the variable with the given @p name from this component.
      * Remove the variable with the given name from this component. If the
-     * name is not found then no variable is removed.
+     * name is not found then throw @c std::out_of_range.
      * @sa addVariable
-     * @overload
      * @param name The name of the variable to remove.
      */
     void removeVariable(const std::string &name);
 
     /**
+     * @brief Remove the variable by the given @p variable pointer from this component.
+     * Remove the variable with the given pointer from this component. If the
+     * variable is not found then no variable is removed.
+     * @sa addVariable
+     *
+     * @overload
+     * @param variable The pointer to the variable to remove.
+     */
+    void removeVariable(const VariablePtr &variable);
+
+    /**
+     * @brief Remove all variables stored in this component.
+     * Clears all variables that have been added to this component.
+     */
+    void removeAllVariables();
```

Comment:
> Are they completely removed or do we have some dangling connections?

There shouldn't be any dangling connections- our `VariableMap` in `Model::doSerialisation` is now only constructed of unique variable equivalence pairs that have equivalence reciprocity (i.e., exist in each other's mEquivalentVariables). 

I'll add doc and tests to make this clearer and double-check. :-)